### PR TITLE
Bug 1870798 - Use fakes in log tests

### PR DESCRIPTION
**Pull request by rahulsainani**
Details in the bug. No implementation or behavior changes.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1870798

### CODE REVIEW

**Comment by Amejia481** [comment]
Thanks for putting this patch!
Could help me to clarify why we need to introduce `FakeLogSink`?  I'm bit confused ,as `LogSink` is an interface, so we don't have any concrete implementation to test, at the moment, their children `AndroidLogSink` and `FenixLogSink` are covered by tests.

Thanks in advance!

**Review by Amejia481 — APPROVED** [review]
I like the 🚈 fast tests :)

**Comment by Amejia481** [comment]
Thanks for the clarification!

File: android-components/components/support/base/src/test/java/mozilla/components/support/base/log/logger/LoggerTest.kt [hunk 1]
```diff
@@ -165,20 +165,15 @@ class LoggerTest {
...
 
-        verify(sink).log(
-            priority = Log.Priority.DEBUG,
-            tag = "WithTag",
-            throwable = null,
-            message = "⇠ testing [10ms]",
-        )
-
-        verifyNoMoreInteractions(sink)
+        Assert.assertEquals(expectedLogs, sink.logs)
```
Comment (MozillaNoah) [inline android-components/components/support/base/src/test/java/mozilla/components/support/base/log/logger/LoggerTest.kt:177]: optional nit: you could directly import `assertEquals` if you wanted

**Review by MozillaNoah — APPROVED** [review]

**Pull request merged**